auth.go

```diff
@@ -0,0 +1,53 @@
+package email
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"net/smtp"
+)
+
+func LoginAuth(username, password, host string) smtp.Auth {
+	return &loginAuth{username, password, host}
+}
+
+// loginAuth is an smtp.Auth that implements the LOGIN authentication mechanism.
+type loginAuth struct {
+	username string
+	password string
+	host     string
+}
+
+func (a *loginAuth) Start(server *smtp.ServerInfo) (string, []byte, error) {
+	if !server.TLS {
+		advertised := false
+		for _, mechanism := range server.Auth {
+			if mechanism == "LOGIN" {
+				advertised = true
+				break
+			}
+		}
+		if !advertised {
+			return "", nil, errors.New("gomail: unencrypted connection")
+		}
+	}
+	if server.Name != a.host {
+		return "", nil, errors.New("gomail: wrong host name")
+	}
```

Comment:
I feel I've seen this implementation before (perhaps when perusing gomail when I've needed to implement this myself) but it does not make sense to me. I'd expect something more like:
```suggestion
	if !server.TLS {
		return "", nil, errors.New("unencrypted connection")
	}
	if server.Name != a.Host {
		return "", nil, errors.New("wrong host name")
	}
	advertised := false
	for _, mechanism := range server.Auth {
		if mechanism == "LOGIN" {
			advertised = true
			break
		}
	}
	if !advertised {
		return "", nil, errors.New("login auth not supported")
	}
```
I don't see how whether it advertises LOGIN has anything to do with an unencrypted connection.  In [a comment in the net/smtp source](https://golang.org/src/net/smtp/auth.go?#L64) it even mentions:

> Note: If TLS is not true, then we can't trust ANYTHING in ServerInfo.

And they use something similar to what I propose above for the smtp.PlainAuth implementation.

Unless I'm missing something?